project_euler/problem_099/sol1.py

```diff
@@ -25,7 +25,8 @@ def solution(data_file: str = "base_exp.txt") -> int:
     largest: float = 0
     result = 0
     for i, line in enumerate(open(os.path.join(os.path.dirname(__file__), data_file))):
-        a, x = list(map(int, line.split(",")))
+        a, x = line.split(",")
+        a, x = int(a), int(x)
```

Comment:
This is confusing and unreadable. If there's only going to be 2 numbers per line, then it's better to use `str.partition`. So, either keep it as it is or update it to use better function :)

Reply:
I don't see how this is unreadable, and I don't see how using `str.partition` would help.
I'll just revert the change...

Reply:
`str.split` splits on every `,` but `str.partition` will only *partition* on the first comma which is the case in the problem's input. Without looking at the input, someone could argue that this code is incorrect as split could return more than 3 element list which would mean that `int([...])` could raise an error.